staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -307,8 +307,8 @@ func (u *Unstructured) GetOwnerReferences() []metav1.OwnerReference {
 
 func (u *Unstructured) SetOwnerReferences(references []metav1.OwnerReference) {
 	var newReferences = make([]map[string]interface{}, 0, len(references))
-	for i := 0; i < len(references); i++ {
-		newReferences = append(newReferences, setOwnerReference(references[i]))
+	for _, reference := range references {
+		newReferences = append(newReferences, setOwnerReference(reference))
```

Comment:
Well, I think using `references[i]` instead of `reference` here (in line 311) is to avoid unnecessary object copy, I'd prefer keep as it is. Or maybe you can change line 310 to 
```go
for i, _ := range references (
```